pkg/cmd/roachtest/cluster.go

```diff
@@ -3141,3 +3142,66 @@ func (c *clusterImpl) GetHostErrorVMs(ctx context.Context, l *logger.Logger) ([]
 	}
 	return allHostErrorVMs, nil
 }
+
+var sideEyeToken, _ = os.LookupEnv("SIDE_EYE_API_TOKEN")
+
+// CaptureSideEyeSnapshot asks the Side-Eye service to take a snapshot of the
+// cockroach processes of this cluster. All errors are logged and swallowed, and
+// the call is a no-op if the SIDE_EYE_API_TOKEN is not in the env. The agents
+// must previously have been installed and started with the cluster's name as
+// the env name.
+func (c *clusterImpl) CaptureSideEyeSnapshot(ctx context.Context, l *logger.Logger) {
```

Comment:
I don't immediately see anything specific to roachtest. Any reason this couldn't be moved to roachprod?

Reply:
#124780 has the implementation of `StartSideEye` in `roachprod`, but also has a wrapper in this `clusterImpl`; I think that for anything that roachtests call, having a method in `clusterImpl` is significantly more ergonomic than calling into roachprod directly.

For `CaptureSideEyeSnapshot`, I'm less sure whether the roachprod code needs to know anything about it. If there was a `roachprod snapshot ...` command (like there apparently is for `roachprod grafana-annotation` for example), then the code would need to be in roachprod. But if roachtest is the only user, then it might be simpler to keep the code here.

But don't listen to me; whatever y'all think.

Reply:
@srosenberg I've revised this diff to only touch roachprod, adding a utility for snapshotting any named env, and a cli cmd calling it for snapshotting an env that matches a cluster's name.